src/views/EventListing/index.js

```diff
@@ -2,36 +2,191 @@ import React from 'react'
+            .then(response => {
+                this.setState({tableData: {
+                    ...this.state.tableData,
+                    events: response.data.data,
+                    count: response.data.meta.count,
+                }})
+            })
+            .finally(() => this.setLoading(true))
+    }
+
+    /**
+     * Handles table column sort changes
+     * @param columnName    The column that should be sorted
+     */
+    handleSortChange = (columnName) => {
+        const {sortBy, sortDirection} = this.state.tableData
+        const updatedSortDirection = getSortDirection(sortBy, columnName, sortDirection)
+        const queryParams = this.getDefaultEventQueryParams()
+        queryParams.setSort(columnName, updatedSortDirection)
+
+        this.setLoading(false)
+
+        fetchEvents(queryParams)
+            .then(response => {
+                this.setState({tableData: {
+                    ...this.state.tableData,
+                    events: response.data.data,
+                    count: response.data.meta.count,
+                    paginationPage: 0,
+                    sortBy: columnName,
+                    sortDirection: updatedSortDirection,
+                }})
+            })
+            .finally(() => this.setLoading(true))
+    }
+
+    /**
+     * Handles table pagination page changes
+     * @param event
+     * @param newPage   The new page number
+     */
+    handlePageChange = (event, newPage) => {
+        const queryParams = this.getDefaultEventQueryParams()
+        queryParams.page = newPage + 1
+
+        this.setLoading(false)
+
+        fetchEvents(queryParams)
+            .then(response => {
+                this.setState({tableData: {
+                    ...this.state.tableData,
+                    events: response.data.data,
+                    count: response.data.meta.count,
+                    paginationPage: newPage,
+                }})
+            })
+            .finally(() => this.setLoading(true))
+    }
+
+    /**
+     * Handles table page size changes
+     * @param   event   Page size selection event data
+     */
+    handlePageSizeChange = (event) => {
+        const pageSize = event.target.value
+        const queryParams = this.getDefaultEventQueryParams()
+        queryParams.page_size = pageSize
+
+        this.setLoading(false)
+
+        fetchEvents(queryParams)
+            .then(response => {
+                this.setState({tableData: {
+                    ...this.state.tableData,
+                    events: response.data.data,
+                    count: response.data.meta.count,
+                    paginationPage: 0,
+                    pageSize: pageSize,
+                }})
+            })
+            .finally(() => this.setLoading(true))
+    }
+
+    /**
+     * Sets the loading state
+     * @param fetchComplete Whether the fetch has completed
+     */
+    setLoading = (fetchComplete) => {
+        const updatedState = {...this.state}
```

Comment:
You only need to specify the member you want to change instead of including the entire state:

```js
this.setState(state => ({
  tableData: {
    ...state.tableData,
    fetchComplete,
  },
}))
```

Also, passing an updater function ensures that the current state that is being referenced is up-to-date.